packages/backend/server/src/plugins/oauth/controller.ts

```diff
@@ -93,43 +104,18 @@ export class OAuthController {
 
     const tokens = await provider.getToken(code);
     const externAccount = await provider.getUser(tokens.accessToken);
-    const user = req.user;
-
-    try {
-      if (!user) {
-        // if user not found, login
-        const user = await this.loginFromOauth(
-          state.provider,
-          externAccount,
-          tokens
-        );
-        const session = await this.auth.createUserSession(
-          user,
-          req.cookies[AuthService.sessionCookieName]
-        );
-        res.cookie(AuthService.sessionCookieName, session.sessionId, {
-          expires: session.expiresAt ?? void 0, // expiredAt is `string | null`
-          ...this.auth.cookieOptions,
-        });
-      } else {
-        // if user is found, connect the account to this user
-        await this.connectAccountFromOauth(
-          user,
-          state.provider,
-          externAccount,
-          tokens
-        );
-      }
-    } catch (e: any) {
-      return res.redirect(
-        this.url.link('/signIn', {
-          redirect_uri: state.redirectUri,
-          error: e.message,
-        })
-      );
-    }
-
-    this.url.safeRedirect(res, state.redirectUri);
+    const user = await this.loginFromOauth(
+      state.provider,
+      externAccount,
+      tokens
+    );
+
+    await this.auth.setCookies(req, res, user.id);
+    res.send({
+      id: user.id,
+      /* @deprecated */
```

Comment:
we have change the api endpoint, is it necessary to keep the old client compatibility code?

Reply:
old clients always use `/desktop-signin` as a transfer